File: frontend/src/types/music.ts
export interface Track {
  id: string
  title: string
  artist: string
  artistId: string
  album: string
  albumId: string
  duration: number          // seconds
  coverUrl: string
  audioUrl: string
  genre: string
  playCount: number
  likeCount: number
  isLiked?: boolean
  createdAt: string
  updatedAt: string
  // Optional Spotify audio features — populated for recommendation-service tracks
  spotifyId?: string
  valence?: number
  energy?: number
  danceability?: number
  tempo?: number
  acousticness?: number
  instrumentalness?: number
  loudness?: number
  speechiness?: number
}

export interface Album {
  id: string
  title: string
  artistId: string
  artist: string
  coverUrl: string
  releaseYear: number
  genre: string
  trackCount: number
  tracks?: Track[]
  createdAt: string
}

export interface Artist {
  id: string
  name: string
  bio?: string
  avatarUrl: string
  coverUrl?: string
  genre: string
  followerCount: number
  albumCount: number
  trackCount: number
  isFollowed?: boolean
  createdAt: string
}

export interface Genre {
  id: string
  name: string
  color: string
  trackCount: number
}

export type TrackSortField = 'title' | 'artist' | 'playCount' | 'duration' | 'createdAt'
export type SortDirection = 'asc' | 'desc'

File: backend/music-service/internal/domain/entity/music.go
package entity

import "time"

type Artist struct {
	ID        string
	Name      string
	Bio       string
	AvatarUrl string
	CreatedAt time.Time
}

type Album struct {
	ID        string
	Title     string
	ArtistID  string
	Artist    string
	Year      int
	CreatedAt time.Time
	CoverUrl  string
}

type Song struct {
	ID         string
	Title      string
	ArtistID   string
	Artist     string
	AlbumID    string
	Album      string
	DurationS  int
	Genre      string
	SearchVec  string
	CreatedAt  time.Time
	UploaderID string
	UpdatedAt  time.Time
	CoverUrl   string
	PreviewUrl string
}
